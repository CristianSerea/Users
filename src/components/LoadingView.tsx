import React from 'react';
import {View, Text, ActivityIndicator, StyleSheet} from 'react-native';
import useColorsTheme from '../constants/colors';
import {dimensions} from '../constants/dimensions';
import {fonts} from '../constants/fonts';
import Strings from '../../strings/strings.json';

const LoadingView = () => {
  const {colors} = useColorsTheme();

  return (
    <View style={styles.container} testID="LoadingView">
      <Text style={fonts.normal}>{Strings.LoadingData}...</Text>
      <ActivityIndicator
        style={styles.activityIndicator}
        color={colors.mainColor}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  activityIndicator: {
    marginTop: dimensions.margin,
  },
});

export default LoadingView;
