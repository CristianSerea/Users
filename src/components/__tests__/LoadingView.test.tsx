import React from 'react';
import {render} from '@testing-library/react-native';
import LoadingView from '../LoadingView';

describe('LoadingView tests', () => {
  it('Test loading view rendering', () => {
    const {getByTestId, toJSON} = render(<LoadingView />);

    expect(getByTestId('LoadingView')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
