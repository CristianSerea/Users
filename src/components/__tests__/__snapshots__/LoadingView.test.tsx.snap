// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingView tests Renders loading view 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="LoadingView"
>
  <Text
    style={
      {
        "fontSize": 15,
        "fontWeight": "600",
      }
    }
  >
    Loading data
    ...
  </Text>
  <ActivityIndicator
    color="#008000"
    style={
      {
        "marginTop": 20,
      }
    }
  />
</View>
`;

exports[`LoadingView tests Test loading view rendering 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="LoadingView"
>
  <Text
    style={
      {
        "fontSize": 15,
        "fontWeight": "600",
      }
    }
  >
    Loading data
    ...
  </Text>
  <ActivityIndicator
    color="#008000"
    style={
      {
        "marginTop": 20,
      }
    }
  />
</View>
`;
